File: 06 Group Anagrams/Group Anagrams.cpp
class Solution
{
public:
    vector<vector<string>> groupAnagrams(vector<string> &strs)
    {
        map<string, vector<string>> mp;
        vector<vector<string>> ans;
        for (int i = 0; i < strs.size(); i++)
        {
            string y = strs[i];
            sort(y.begin(), y.end());
            mp[y].push_back(strs[i]);
        }

        for (map<string, vector<string>>::iterator it = mp.begin(); it != mp.end(); it++)
        {
            ans.push_back(it->second);
        }
        return ans;
    }
};

File: 06 Group Anagrams/Group Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
    let map = new Map();
    strs.forEach(element => {
      let str = element.split('').sort().join('');
      let temp = [];
      if (map.has(str))
        temp = map.get(str);
      temp.push(element);
      map.set(str, temp);
    });
    return Array.from(map.values());
  };